x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtRealm.java

```diff
@@ -401,8 +406,14 @@ public void authenticate(final AuthenticationToken authenticationToken, final Ac
                 // Intercept the role mapper listener response to log the resolved roles here. Empty is OK.
                 final String[] rolesArray = rolesSet.toArray(new String[0]);
                 final User user = new User(principal, rolesArray, null, null, userData.getMetadata(), true);
-                final String rolesString = Arrays.toString(rolesArray);
-                LOGGER.debug("Realm [" + super.name() + "] mapped roles " + rolesString + " for principal=[" + principal + "].");
+                LOGGER.debug(
+                    () -> new ParameterizedMessage(
+                        "Realm [{}] mapped roles [{}] for principal=[{}].",
+                        super.name(),
+                        Strings.arrayToCommaDelimitedString(rolesArray),
```

Comment:
Nit. Why Strings.arrayToCommaDelimitedString? Everywhere else in JWT uses String.join() for collections and arrays.

Reply:
No particular reason in this case.
In general I default to  `Strings.arrayToCommaDelimitedString` because the methods on `Strings` support Objects but `String` only supports `CharSequence`. There are enough cases where we have other object types (Realm ids, longs, etc) that I just default to the method that always works.

I can change it. Overall Elasticsearch seems to have a slight preference for the version on `Strings`
(192 total usage of the various versions there) vs `String.join` (102 usages), but I don't think there's a compelling reason to go in either direction.